lib/dry/system/auto_registration_config.rb

```diff
@@ -0,0 +1,22 @@
+module Dry
+  module System
+    # @api private
+    AutoRegistraionConfig = Struct.new(:instance, :exclude) do
+      def instance(&block)
+        @instance = block
+      end
+
+      def exclude(&block)
+        @exclude = block
+      end
+
+      def call(method, component)
+        if proc = instance_variable_get("@#{method}")
+          proc.(component)
+        else
+          nil
+        end
+      end
```

Comment:
If we went along the lines of the example I left you in my last comment, I don't think we'd need this `#call`. Instead, we could just send the `#instance` or `#exclude` messages directly, e.g. `registration_config.exclude.(component)`, which feels a little nicer.